lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -435,6 +435,26 @@ export default class BitcoinProcessor {
 
     const [transactions, lastBlockSeen] = await this.getTransactionsSince(since, lastProcessedBlock.height);
 
+    // Add normalizedFee to transactions because internal to bitcoin, normalizedFee live in blockMetadata and have to be joined by block height
+    // with transactions to get per transaction normalizedFee.
+    if (transactions.length !== 0) {
+      const inclusiveFirstBlockHeight = transactions[0].transactionTime;
+      const exclusiveLastBlockHeight = transactions[transactions.length - 1].transactionTime + 1;
+      const blockMetaData = await this.blockMetadataStore.get(inclusiveFirstBlockHeight, exclusiveLastBlockHeight);
+      let blockMetadataIndex = 0;
+      for (const transaction of transactions) {
+        while (transaction.transactionTime !== blockMetaData[blockMetadataIndex].height) {
+          blockMetadataIndex++;
+          // This means that it has reached the end of block metadata and still cannot find the data for the current transaction.
+          // This can happen when a block reorg happens, and some block metadata are getting deleted.
+          if (blockMetadataIndex === blockMetaData.length) {
+            throw new RequestError(ResponseStatus.ServerError, ErrorCode.BitcoinBlockMetadataNotFound);
+          }
+        };
+        transaction.normalizedTransactionFee = blockMetaData[blockMetadataIndex].normalizedFee;
+      }
```

Comment:
Optional: consider a  less performant but easier to read implementation that involves 2 un-nested loops: first loop constructs a map, 2nd loop uses the map.